test/header_parser_tests/expected_bindings/_expected_objc_interface_bindings.dart

```diff
@@ -0,0 +1,980 @@
+    _sel_alloc ??= _registerName(_lib, "alloc");
+    final _ret = _lib.objc_msgSend_0(_class!, _sel_alloc!);
+    return NSObject._(_ret, _lib);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_copy;
+  NSObject copy() {
+    _sel_copy ??= _registerName(_lib, "copy");
+    final _ret = _lib.objc_msgSend_2(_id, _sel_copy!);
+    return NSObject._(_ret, _lib);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_mutableCopy;
+  NSObject mutableCopy() {
+    _sel_mutableCopy ??= _registerName(_lib, "mutableCopy");
+    final _ret = _lib.objc_msgSend_3(_id, _sel_mutableCopy!);
+    return NSObject._(_ret, _lib);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_copyWithZone;
+  static NSObject copyWithZone(NativeLibrary _lib, ffi.Pointer<_NSZone> zone) {
+    _class ??= _getClass(_lib, "NSObject");
+    _sel_copyWithZone ??= _registerName(_lib, "copyWithZone:");
+    final _ret = _lib.objc_msgSend_4(_class!, _sel_copyWithZone!, zone);
+    return NSObject._(_ret, _lib);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_mutableCopyWithZone;
+  static NSObject mutableCopyWithZone(
+      NativeLibrary _lib, ffi.Pointer<_NSZone> zone) {
+    _class ??= _getClass(_lib, "NSObject");
+    _sel_mutableCopyWithZone ??= _registerName(_lib, "mutableCopyWithZone:");
+    final _ret = _lib.objc_msgSend_5(_class!, _sel_mutableCopyWithZone!, zone);
+    return NSObject._(_ret, _lib);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_instancesRespondToSelector;
+  static int instancesRespondToSelector(
+      NativeLibrary _lib, ffi.Pointer<ObjCSel> aSelector) {
+    _class ??= _getClass(_lib, "NSObject");
+    _sel_instancesRespondToSelector ??=
+        _registerName(_lib, "instancesRespondToSelector:");
+    return _lib.objc_msgSend_6(
+        _class!, _sel_instancesRespondToSelector!, aSelector);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_conformsToProtocol;
+  static int conformsToProtocol(NativeLibrary _lib, NSObject protocol) {
+    _class ??= _getClass(_lib, "NSObject");
+    _sel_conformsToProtocol ??= _registerName(_lib, "conformsToProtocol:");
+    return _lib.objc_msgSend_7(_class!, _sel_conformsToProtocol!, protocol._id);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_methodForSelector;
+  IMP methodForSelector(ffi.Pointer<ObjCSel> aSelector) {
+    _sel_methodForSelector ??= _registerName(_lib, "methodForSelector:");
+    return _lib.objc_msgSend_8(_id, _sel_methodForSelector!, aSelector);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_instanceMethodForSelector;
+  static IMP instanceMethodForSelector(
+      NativeLibrary _lib, ffi.Pointer<ObjCSel> aSelector) {
+    _class ??= _getClass(_lib, "NSObject");
+    _sel_instanceMethodForSelector ??=
+        _registerName(_lib, "instanceMethodForSelector:");
+    return _lib.objc_msgSend_9(
+        _class!, _sel_instanceMethodForSelector!, aSelector);
+  }
+
+  static ffi.Pointer<ObjCSel>? _sel_forwardingTargetForSelector;
+  NSObject forwardingTargetForSelector(ffi.Pointer<ObjCSel> aSelector) {
```

Comment:
Most of those methods are static methods inherited from NSObject. Dart classes don't inherit static methods, so I have to copy them across manually. This is important because this is how the constructors work (new/alloc/init).

We can definitely filter these methods. I haven't done the filter config for instances/methods yet (gonna do it in a follow up PR), but I wonder if a design just like all the other filters would be sufficient filtering. Maybe we don't need to special case these methods at all?

Btw, instVar is an instance variable, and effectively private, so I'm ignoring it intentionally. @property is for public fields

Reply:
> I haven't done the filter config for instances/methods yet (gonna do it in a follow up PR)

I'd prefer having it in this PR so it's easier to see the history of supported features on the generated files to see what we support. For example if we end up doing a `package:ffigen` release somewhere for non ObjectiveC features. I'd like the ObjectiveC part to be in a conservative feature set. E.g. if we start hiding those symbols by default later to reduce clutter that would be a breaking change on the generated API.

Reply:
> Most of those methods are static methods inherited from NSObject. Dart classes don't inherit static methods, so I have to copy them across manually. This is important because this is how the constructors work (new/alloc/init).

Looks like we just use them in the implementation of alloc etc, which means we should not expose them to our users.

```dart
  static ffi.Pointer<ObjCSel>? _sel_new1;
  static Foo new1(NativeLibrary _lib) {
    _class ??= _getClass(_lib, "Foo");
    _sel_new1 ??= _registerName(_lib, "new");
    final _ret = _lib.objc_msgSend_0(_class!, _sel_new1!);
    return Foo._(_ret, _lib);
  }

  static ffi.Pointer<ObjCSel>? _sel_allocWithZone;
  static Foo allocWithZone(NativeLibrary _lib, ffi.Pointer<_NSZone> zone) {
    _class ??= _getClass(_lib, "Foo");
    _sel_allocWithZone ??= _registerName(_lib, "allocWithZone:");
    final _ret = _lib.objc_msgSend_1(_class!, _sel_allocWithZone!, zone);
    return Foo._(_ret, _lib);
  }

  static ffi.Pointer<ObjCSel>? _sel_alloc;
  static Foo alloc(NativeLibrary _lib) {
    _class ??= _getClass(_lib, "Foo");
    _sel_alloc ??= _registerName(_lib, "alloc");
    final _ret = _lib.objc_msgSend_0(_class!, _sel_alloc!);
    return Foo._(_ret, _lib);
  }
```

We can make `objc_msgSend_1` and the others private in Dart.

And `forwardingTargetForSelector` isn't used anywhere.